frontend/javascripts/admin/admin_rest_api.ts

```diff
@@ -2377,6 +2383,49 @@ export function getAgglomerateSkeleton(
   );
 }
 
+export async function getAgglomeratesForSegmentsFromDatastore<T extends number | bigint>(
+  dataStoreUrl: string,
+  datasetId: APIDatasetId,
+  layerName: string,
+  mappingId: string,
+  segmentIds: Array<T>,
+): Promise<Mapping> {
+  const segmentIdBuffer = serializeProtoListOfLong<T>(segmentIds);
+  const listArrayBuffer: ArrayBuffer = await doWithToken((token) =>
+    Request.receiveArraybuffer(
+      `${dataStoreUrl}/data/datasets/${datasetId.owningOrganization}/${datasetId.name}/layers/${layerName}/agglomerates/${mappingId}/agglomeratesForSegments?token=${token}`,
+      {
+        method: "POST",
+        body: segmentIdBuffer,
+        headers: {
+          "Content-Type": "application/octet-stream",
+        },
+      },
+    ),
+  );
+  // @ts-ignore
+  return new Map(_.zip(segmentIds, parseProtoListOfLong(listArrayBuffer)));
+}
+
+export async function getAgglomeratesForSegmentsFromTracingstore<T extends number | bigint>(
+  tracingStoreUrl: string,
+  tracingId: string,
+  segmentIds: Array<T>,
+): Promise<Mapping> {
+  const mappingObject: Record<string, number> | Record<string, bigint> = await doWithToken(
+    (token) =>
+      Request.sendJSONReceiveJSON(
+        `${tracingStoreUrl}/tracings/mapping/${tracingId}/agglomeratesForSegments?token=${token}`,
+        {
+          data: segmentIds,
```

Comment:
good question! i don't really know why getAgglomeratesForSegmentsFromDatastore and getAgglomeratesForSegmentsFromTracingstore differ in that regard. maybe @daniel-wer and @fm3 know something?

Reply:
The protobuf one was originally designed as a backend-internal route (between datastore and tracingstore, I think). Since this PR was originally a hackathon project, Daniel and I agreed that it wasn’t necessary to unify it, and the frontend could now just use the existing datastore route.

This is of course something we can discuss. In theory, we could change a lot of high-volume APIs to protobuf and probably save some performance. However, it also means that the requests aren’t human-readable which may make debugging harder. I’m not sure about a good general rule here.

If you would prefer it, I can change this route to accept and send protobuf too…? Or go the other way round and offer a json one for the datastore. We can also defer this discussion to a new issue.

Protobuf-generated classes tend to be a bit cumbersome to work with (at least in the backend, as we cannot control their inheritance hierarchy or add custom functions to them), while storage and de/serialization is definitely more efficient than for json.

Reply:
I see 3 options:
1) Pure JSON with numbers (constrained to 53 bit ids which would be fine for this PR)
2) Send/receive array buffers (similar to how we transmit bucket data). Either, we use Float64Array so that we have 53 bit integers again. Or, we use BigUint64Array (however, I'm not sure whether it's easy to encode/decode in the back-end).
3) Protobuf. It works, I guess, but I also see the hassle of it. If (2) is feasible, I'd prefer it in the long run.

I think, I'd favor (1) for now. but maybe let's discuss this together in a call next week (hopefully with @daniel-wer :)).

Reply:
for the record: we went for protobuf for now. bigint support wasn't really straightforward with this, though (I only went for 53 bit compatibility now). we can revise this later.